### PR TITLE
feat(EditGroup): Add and Remove participants in group

### LINKED ISSUE #545: Groups - add ability to leave a group and add more users
**Task:** The user should be able to leave the group chat, using the context menu or having a button to leave maybe have some thing in the title bar. 

We also want to be able to add more people to the group 

**Screenshot:**
Screenshot 2023-03-28 at 4 02 04 PM

### PR DESCRIPTION
**Pull request by lgmarchi**
### 1. Add participants in a group 

https://user-images.githubusercontent.com/63157656/232572772-a8d54412-bd80-4537-83d0-96f9e561dbf8.mov

### 2. Remove participants from a group

https://user-images.githubusercontent.com/63157656/232572899-aea38733-f088-4160-8d6a-35e0de8da416.mov

- 

### Which issue(s) this PR fixes 🔨

- Resolve #545

### CODE REVIEW

**Comment by phillsatellite** [comment]
@lgmarchi  Hey my friend! Getting this error on Windows can you verify?
 
image

**Comment by lgmarchi** [comment]
> @lgmarchi Hey my friend! Getting this error on Windows can you verify?
> 
> image

Verified and fixed my friend @phillsatellite

**Comment by phillsatellite** [comment]
Should we only have this feature appear when you are the creator of the group chat? It seems a little confusing for someone who is not a creator of the group

https://user-images.githubusercontent.com/93608357/232615205-3531759a-bdcc-4d79-a571-1d0adb311c91.mov

**Comment by lgmarchi** [comment]
> Should we only have this feature appear when you are the creator of the group chat? It seems a little confusing for someone who is not a creator of the group
> 
>  Screen.Recording.2023-04-17.at.5.30.44.PM.mov

Ooh it is wrong, it should appear, you are not group creator, I will fix that,  thanks man

**Review by phillsatellite — APPROVED** [review]

File: common/src/state/mod.rs [hunk 1]
```diff
@@ -454,6 +454,22 @@ impl State {
...
                 }
             }
+            MessageEvent::RecipientAdded {
+                conversation,
+                identity,
+            } => {
+                self.identities.insert(identity.did_key(), identity);
+                if let Some(chat) = self.chats.all.get_mut(&conversation.id()) {
+                    chat.participants =
+                        HashSet::from_iter(conversation.recipients().iter().cloned());
```
Comment (dariusc93) [inline common/src/state/mod.rs]:
```suggestion
                        HashSet::from_iter(conversation.recipients());
```

File: common/src/state/mod.rs [hunk 2]
```diff
@@ -454,6 +454,22 @@ impl State {
...
+                }
+            }
+            MessageEvent::RecipientRemoved { conversation } => {
+                if let Some(chat) = self.chats.all.get_mut(&conversation.id()) {
+                    chat.participants =
+                        HashSet::from_iter(conversation.recipients().iter().cloned());
```
Comment (dariusc93) [inline common/src/state/mod.rs]:
```suggestion
                        HashSet::from_iter(conversation.recipients());
```

File: ui/src/components/chat/edit_group.rs [hunk 3]
```diff
@@ -0,0 +1,372 @@
...
+            while let Some(cmd) = rx.next().await {
+                match cmd {
+                    ChanCmd::AddParticipants => {
+                        let recipients: Vec<DID> =
+                            selected_friends.current().iter().cloned().collect();
+                        let (tx, rx) = oneshot::channel();
+                        if let Err(e) =
+                            warp_cmd_tx.send(WarpCmd::RayGun(RayGunCmd::AddGroupParticipants {
+                                conv_id,
+                                recipients: recipients.clone(),
```
Comment (dariusc93) [inline ui/src/components/chat/edit_group.rs]:
```suggestion
                                recipients,
```

File: ui/src/components/chat/edit_group.rs [hunk 4]
```diff
@@ -0,0 +1,372 @@
...
+                        }
+                    }
+                    ChanCmd::RemoveParticipants => {
+                        let recipients: Vec<DID> =
+                            selected_friends.current().iter().cloned().collect();
+                        let (tx, rx) = oneshot::channel();
+                        if let Err(e) =
+                            warp_cmd_tx.send(WarpCmd::RayGun(RayGunCmd::RemoveGroupParticipants {
+                                conv_id,
+                                recipients: recipients.clone(),
```
Comment (dariusc93) [inline ui/src/components/chat/edit_group.rs]:
```suggestion
                                recipients,
```

**Review by sdwoodbury — APPROVED** [review]

**Pull request merged**